src/Util/Lock/SemaphoreLockDriver.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Friendica\Util\Lock;
+
+class SemaphoreLockDriver implements ILockDriver
```

Comment:
Please add a constructor throwing an Exception if the required `sem_get` function doesn't exist. You can handle the exception by providing a fallback driver in `Core\Lock`.